src/Markdig/Renderers/TextRendererBase.cs

```diff
@@ -174,10 +174,12 @@ public void PushIndent(string[] lineSpecific)
 
     public void PopIndent()
     {
-        // TODO: Check
-        indents.RemoveAt(indents.Count - 1);
+        if (this.indents.Count > 0)
```

Comment:
This is not better, because it hides the fact that there are more PopIndent than push, which is a bug, so this PR hides bugs. The `TODO: Check` was to say "provide a better error message" instead of failing with list removal.

Reply:
Ah, my bad sorry. I've added an exception to be thrown when it is empty.

Reply:
(Sorry for the late response)